### DISCUSSION TITLE
When I rotate a page to 270 degrees and want to access it later from appending, any page rotated 270 is adding +1 to the index

### DISCUSSION THREAD

**Discussion by Begimkins** [pymupdf/PyMuPDF#3964] [Looking for help] [answered]
I have a Contents Appendix and whenever I rotate pages, export them and try to access those pages that are 270 degree rotation, the link is giving me the next page. 

For example:
Page 15 - rotated 270 
Page 16 - rotation 90 

If I press Page 15 from my Contents list, the link goes to page 16. 
if both page 15 and page 16 are rotated 270, it goes to page 17.

How do I fix that?

**Comment by JorjMcKie** [comment]
Please be more verbose:
What "contents list" are you talking about? What does "pressing" a page mean anyway? What link?
Etc.

**Reply by Begimkins** [reply]
This is what I mean by Content List: Table of Contents: 
<img width="429" alt="Screenshot 2024-10-18 at 1 08 54 PM" src="https://github.com/user-attachments/assets/b6a76ea9-84eb-4878-8487-db9d4020dc15">

These are links, I use  `uri = link["uri"]` to create links. 

Whenever I try to click on a page and if the page is rotated at 270 degrees, my pdf viewer moves to the next page available

**Reply by Begimkins** [reply]
this is my code to add page numbers: 
```
 if page.rotation == 0:
            x_pos = page.rect.width - 75
            y_pos = page.rect.height - 25
        elif page.rotation == 90:
            x_pos = page.rect.height - 25
            y_pos = 75
        elif page.rotation == 180:
            x_pos = 75
            y_pos = 25
        elif page.rotation == 270:
            x_pos = 25
            y_pos = page.rect.width - 75
```

**Comment by JorjMcKie** [comment]
I don't believe that your problem has to do with page rotations.
So, you seem to add a TOC. What cannot determine from your explanations:
* Are you creating page(s) in your document to output the indicated text lines of the TOC or are you making a TOC **_list_** to be stored in the PDF via `doc.set_toc(list)`?

If you **_write_** a TOC as text then that page (and any subsequent ones that may be necessary) invalidate the the page numbers coming later: If you add a TOC page (say: it will become page number 2), then the previous page number 2 will now be 3, and so on.

Also don't see why you use link kind `LINK_URI` and not `LINK_GOTO`. Or are you pointing to separate PDF's with your TOC? If so, you should use `LINK_GOTOR`.

**Reply by Begimkins** [reply]
Hmm, thank you. To answer your question, the TOC content is directly rendered as visible PDF pages, with interactive links adjusted to point to the correct pages and use LINK_GOTOR as well, by doing this:` link["kind"] = fitz.LINK_GOTO`

**Reply by Begimkins** [reply]
I dont see my page numbers only when the rotation as at 270 degrees, so i thought that was the issue

**Reply by JorjMcKie** [reply]
So you want to put a page number on the target pages ... in those other / referenced documents?

**Reply by JorjMcKie** [reply, marked as answer]
In your link dictionary, you  must specify the target page number and the "to" point. Depending on where on the page you want that to be, the computation of the correct value may be a little awkward.

But if all fails, you can simply specify `"to": pymupdf.Point(0, 0)` which will point to the target page's bottom left. In any case, you should definitely see the target page popping up.

**Reply by Begimkins** [reply]
Thanks, I will do that!
